src/main/java/com/github/webicitybrowser/webicity/renderer/backend/html/tags/TagAction.java

```diff
@@ -0,0 +1,36 @@
+package com.github.webicitybrowser.webicity.renderer.backend.html.tags;
+
+import com.github.webicitybrowser.spec.css.parser.CSSParser;
+import com.github.webicitybrowser.spec.css.parser.tokenizer.CSSTokenizer;
+import com.github.webicitybrowser.spec.css.parser.tokens.Token;
+import com.github.webicitybrowser.spec.css.rule.CSSRule;
+import com.github.webicitybrowser.spec.css.rule.CSSRuleList;
+import com.github.webicitybrowser.spec.dom.node.Element;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashMap;
+
+public interface TagAction {
+
+	CSSRuleList getCSSRuleList(Element element);
```

Comment:
The logic for adding the rule list to the document should be inside the tag action

Instead, I would suggest having the method be `default void onTagParsed(Element element) {}`